### PR TITLE
Fixed spec at Vector module

### LINKED ISSUE #6: A few bad @spec at Vector module

### PR DESCRIPTION
**Pull request by ondrej-tucek**
Resolve #6

### CODE REVIEW

File: lib/matrix_reloaded/vector.ex [hunk 1]
```diff
@@ -48,7 +48,7 @@ defmodule MatrixReloaded.Vector do
 
   """
 
-  @spec col(pos_integer, number) :: [t()]
```
Comment (s-m-i-t-a) [inline lib/matrix_reloaded/vector.ex]: What is wrong here?

File: lib/matrix_reloaded/vector.ex [hunk 2]
```diff
@@ -65,7 +65,7 @@ defmodule MatrixReloaded.Vector do
       [1, 2, 3]
 
   """
-  @spec transpose(t()) :: t()
```
Comment (s-m-i-t-a) [inline lib/matrix_reloaded/vector.ex]: What is wrong here?

File: lib/matrix_reloaded/vector.ex [hunk 3]
```diff
@@ -260,7 +260,7 @@ defmodule MatrixReloaded.Vector do
 
   """
 
-  @spec mult_by_num(t(), number) :: t()
```
Comment (s-m-i-t-a) [inline lib/matrix_reloaded/vector.ex]: What is wrong here?

File: lib/matrix_reloaded/vector.ex (hunk 2 above)
Comment (s-m-i-t-a) [reply lib/matrix_reloaded/vector.ex]: A better solution could be the definition of a vector as a column vector and vector, eg. `@type column() :: [t()]` (or better name `@type transposed() :: [t()]` ???) and `@type t() :: [number]`. Or derive type of the vector from the matrix. The change of type is quite unclear and complicates the readability.

Comment (ondrej-tucek) [reply lib/matrix_reloaded/vector.ex]: Oki, that sounds better. I'll fix it.

**Pull request merged**